### ISSUE TITLE
AxiosHeaders toJSON

### ISSUE THREAD

**Issue by askmewhat** [axios/axios#5680]
### Describe the bug

The judgment of "value !== false" inside the toJSON method of the AxiosHeaders class, is it necessary? If I add a header with a value of false, it will be automatically deleted. Is this reasonable?

### Code snippet

```js
utils.forEach(this, (value, header) => {
      value != null && value !== false && (obj[header] = asStrings && utils.isArray(value) ? value.join(', ') : value);
    });
```

### Expected behavior

Remove the judgment of "value !== false".

### Axios Version

1.x

### Adapter Version

xhr

### Browser

Chrome

### Browser Version

112.0.5615.138

### Node.js Version

16.20.0

### OS

Win 11

**Comment by DigitalBrainJS** [comment]
Yes, the false value is taken into account by the rewrite option in the AxiosHeader methods. Therefore, such a value will not be overwritten unless the overwrite option is explicitly specified in the set method. Axios logic relies on this to allow the user to opt out of installing unwanted and/or default headers.
